api/v1alpha1/costmanagement_types.go

```diff
@@ -206,19 +210,47 @@ type CloudDotRedHatSourceStatus struct {
 // PrometheusStatus defines the status for querying prometheus
 type PrometheusStatus struct {
 
-	// PrometheusConnected is a field of CostManagementStatus to represent if cost-management is connected to prometheus
-	PrometheusConnected *bool `json:"prometheus_connected,omitempty"`
+	// PrometheusConfigured is a field of CostManagementStatus to represent if cost-management is configured to connect to prometheus.
+	PrometheusConfigured bool `json:"prometheus_configured"`
+
+	// ConfigError is a field of CostManagementStatus to represent errors during prometheus configuration.
+	ConfigError string `json:"configuration_error,omitempty"`
+
+	// PrometheusConnected is a field of CostManagementStatus to represent if prometheus can be queried.
+	PrometheusConnected bool `json:"prometheus_connected"`
+
+	// ConnectionError is a field of CostManagementStatus to represent errors during prometheus test query.
+	ConnectionError string `json:"prometheus_connection_error,omitempty"`
 
-	// LastQueryStartTime is a field of CostManagementStatus to represent the last time queries were started
+	// LastQueryStartTime is a field of CostManagementStatus to represent the last time queries were started.
 	// +nullable
 	LastQueryStartTime metav1.Time `json:"last_query_start_time,omitempty"`
 
-	// LastQuerySuccessTime is a field of CostManagementStatus to represent the last time queries were successful
+	// LastQuerySuccessTime is a field of CostManagementStatus to represent the last time queries were successful.
 	// +nullable
 	LastQuerySuccessTime metav1.Time `json:"last_query_success_time,omitempty"`
 
-	// SvcAddress is the internal thanos-querier address
-	SvcAddress string `json:"address,omitempty"`
+	// SvcAddress is the internal thanos-querier address.
+	SvcAddress string `json:"service_address,omitempty"`
```

Comment:
should we make the SvcAddress fields consistent. In the spec it is `address` and in the status it is `service_address` ?